web-server/pages/api/internal/version.ts

```diff
@@ -103,25 +104,61 @@ async function fetchDockerHubTags(): Promise<
   }));
 }
 
+async function fetchLatestGitHubCommit(): Promise<GitHubCommit> {
+  const apiUrl = `https://api.github.com/repos/${githubOrgName}/${githubRepoName}/commits`;
+  const response = await axios.get<GitHubCommit[]>(apiUrl);
+  const latestCommit = response.data[0];
+  return latestCommit;
+}
+
+function isUpdateAvailable(
+  localVersionInfo: ProjectVersionInfo,
+  dockerLatestRemoteTag: TagCompressed,
+  githubLatestCommit: GitHubCommit
+): boolean {
+  const env = process.env.NEXT_PUBLIC_APP_ENVIRONMENT;
+
+  if (env == 'development') {
+    const localBuildDate = new Date(localVersionInfo.current_build_date);
+    const latestRemoteCommitDate = new Date(
+      githubLatestCommit.commit.author.date
+    );
+
+    return latestRemoteCommitDate > localBuildDate;
+  } else {
+    const localBuildDate = new Date(localVersionInfo.current_build_date);
+    const latestRemoteDate = new Date(dockerLatestRemoteTag.last_updated);
+    return latestRemoteDate > localBuildDate;
+  }
+}
+
 async function checkNewImageRelease(): Promise<CheckNewVersionResponse> {
-  const versionInfo = readVersionFile();
-  const localDate = new Date(versionInfo.docker_image_build_date);
-  const remoteTags = await fetchDockerHubTags();
+  const versionInfo = getProjectVersionInfo();
 
-  remoteTags.sort(
+  const dockerRemoteTags = await fetchDockerHubTags();
+  const githubLatestCommit = await fetchLatestGitHubCommit();
+
+  dockerRemoteTags.sort(
     (a, b) =>
       new Date(b.last_updated).getTime() - new Date(a.last_updated).getTime()
   );
-  const latestTag = remoteTags[0];
+  const latestTag = dockerRemoteTags[0];
   const latestRemoteDate = new Date(latestTag.last_updated);
```

Comment:
this implies that `dockerRemoteTags` can never be empty, and `dockerRemoteTags[0]` always exists. if not, use optional chaining
```
const latestRemoteDate = new Date(latestTag?.last_updated);
```

[optional]
we often use `head` from `ramda` to get the first element of the array
```
const latestTag = head(dockerRemoteTags);
```

Reply:
If it's possible for `latestTag` to actually be undefined (as implied by the optional chain), then don't pass it to `new Date(...)` like that.

`new Date(with a possibly undefined value)` will throw an error. Handle for this.